File: app/src/main/java/de/michael/filebinmobile/adapters/UploadUrlAdapter.java
package de.michael.filebinmobile.adapters;

import android.app.Activity;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import de.michael.filebinmobile.R;

import static android.content.Context.CLIPBOARD_SERVICE;

public class UploadUrlAdapter extends SimpleDataAdapter<UploadUrlViewHolder, String> {

    public UploadUrlAdapter(Activity activity) {
        super(activity);
    }

    @Override
    public UploadUrlViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_upload_url, null);

        return new UploadUrlViewHolder(view);
    }

    @Override
    public void onBindViewHolder(UploadUrlViewHolder holder, int position) {

        String url = getData().get(position);

        holder.txtUploadUrl.setText(url);

        holder.btnClipboardUrl.setOnClickListener(view -> {
            ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(CLIPBOARD_SERVICE);
            ClipData clip = ClipData.newPlainText("FileBin mobile url", url);
            if (clipboard != null) {
                clipboard.setPrimaryClip(clip);
                Toast.makeText(getActivity(), "Url copied to clipboard", Toast.LENGTH_SHORT).show();
            }
        });

    }

    @Override
    public int getItemCount() {
        return getData().size();
    }
}

File: app/src/main/java/de/michael/filebinmobile/adapters/UploadUrlAdapter.kt
package de.michael.filebinmobile.adapters

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Context.CLIPBOARD_SERVICE
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import de.michael.filebinmobile.R
import kotlinx.android.synthetic.main.list_item_upload_url.view.*

class UploadUrlAdapter(val context: Context) : SimpleDataAdapter<UploadUrlViewHolder, String>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UploadUrlViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_upload_url, null)

        return UploadUrlViewHolder(view, context)
    }
}

class UploadUrlViewHolder(itemView: View, val context: Context) : AbstractViewHolder<String>(itemView) {
    override fun bindItem(item: String, removeItem: (Int) -> Unit, pos: Int, onClick: (String) -> Boolean) {
        itemView.txtUploadUrl.text = item
        itemView.btnClipboardUrl.setOnClickListener {

            val clipboard = context.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText("FileBin mobile url", item)

            clipboard.primaryClip = clip
            Toast.makeText(context, "Url copied to clipboard", Toast.LENGTH_SHORT).show()
        }
    }
}